test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/ListPackageTests.cs

```diff
@@ -156,6 +163,123 @@ public void BasicListPackage_OutputFormat_BadInput_Parsing_Fails(string outputFo
+        {
+            // Arrange
+            using var pathContext = new SimpleTestPathContext();
+
+            var packageA100 = new SimpleTestPackageContext("A", "1.0.0");
+            var packageB100 = new SimpleTestPackageContext("B", "1.0.0");
+
+            await SimpleTestPackageUtility.CreatePackagesAsync(
+                    pathContext.PackageSource,
+                    packageA100,
+                    packageB100);
+
+            var projectA = SimpleTestProjectContext.CreateNETCore("ProjectA", pathContext.SolutionRoot, "net6.0");
+            var projectB = SimpleTestProjectContext.CreateNETCore("ProjectB", pathContext.SolutionRoot, "net6.0");
+
+            projectA.AddPackageToAllFrameworks(packageA100);
+            projectB.AddPackageToAllFrameworks(packageB100);
+
+            var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+            solution.Projects.Add(projectA);
+            solution.Projects.Add(projectB);
+            solution.Create(pathContext.SolutionRoot);
+
+            SimpleTestSettingsContext.RemoveSource(pathContext.Settings.XML, "source");
+
+            using var mockServer = new FileSystemBackedV3MockServer(pathContext.PackageSource, isPrivateFeed: true);
+            mockServer.Start();
+            pathContext.Settings.AddSource(sourceName: "private-source", sourceUri: mockServer.ServiceIndexUri, allowInsecureConnectionsValue: bool.TrueString);
+
+            var mockedCredentialService = new Mock<ICredentialService>();
+            var expectedCredentials = new NetworkCredential("user", "password1");
+            SetupCredentialServiceMock(mockedCredentialService, expectedCredentials, new Uri(mockServer.ServiceIndexUri));
+            HttpHandlerResourceV3.CredentialService = new Lazy<ICredentialService>(() => mockedCredentialService.Object);
+
+            // List package command requires restore to be run before it can list packages.
+            await RestoreProjectsAsync(pathContext, projectA, projectB);
+
+            // Act
+            ListPackageCommandRunner listPackageCommandRunner = new();
+            var packageRefArgs = new ListPackageArgs(
+                                        path: Path.Combine(pathContext.SolutionRoot, "solution.sln"),
+                                        packageSources: [new(mockServer.ServiceIndexUri)],
+                                        frameworks: ["net6.0"],
+                                        reportType: ReportType.Vulnerable,
+                                        renderer: new ListPackageConsoleRenderer(),
+                                        includeTransitive: false,
+                                        prerelease: false,
+                                        highestPatch: false,
+                                        highestMinor: false,
+                                        logger: NullLogger.Instance,
```

Comment:
test: If you pass a `TestLogger` here, you can then show all the messages from there in line 223, if case the assertion fails.

Reply:
This is precisely what I did in the commit at https://github.com/NuGet/NuGet.Client/pull/5571/commits/f8f381b658fc390292525ed18be048c6d3eb33c0 to troubleshoot why the test failed on Linux. Then, I realized that there were other tests using the mock server that also failed on Linux and had the `Platform Fact Windows` configuration set. I can implement the same changes in the next commit.

Reply:
Fixed in the latest commit.